packages/puppeteer-core/src/common/bidi/Sandbox.ts

```diff
@@ -68,6 +70,19 @@ export class Sandbox implements RealmBase {
   constructor(context: Realm, timeoutSettings: TimeoutSettings) {
     this.#realm = context;
     this.#timeoutSettings = timeoutSettings;
+
+    // TODO: we should track the correct realm
+    // Temporary fix
+    // This only works for the main context realm
+    this.#realm.on(Bidi.ChromiumBidi.Script.EventNames.RealmCreated, () => {
```

Comment:
There should be a better way to handle this, still looking over it, as it was part of an old stash,

Reply:
Left a comment, and we do the same thing as per PuppeteerUtils.